src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -41,6 +42,7 @@ public Command parseCommand(String userInput) throws ParseException {
         }
 
         final String commandWord = matcher.group("commandWord");
+        final String flag = matcher.group("flag"); // returns a whitespace, followed by the tag e.g. " -f".
         final String arguments = matcher.group("arguments");
```

Comment:
In addition, I would like to request for you to please parse arguments into a proper format before passing it onto the CommandParser classes, or else this logic will have to be repeated in all the commandparser classes. LMK if you would prefer this done in a separate PR or by someone else

Reply:
Could I clarify what you mean by a proper format?

Reply:
Sorry, please ignore - this should be done in another PR